lab_1_keywords_tfidf/main.py

```diff
@@ -99,7 +151,25 @@ def calculate_tfidf(term_freq: dict[str, float], idf: dict[str, float]) -> Optio
 
     In case of corrupt input arguments, None is returned
     """
-    pass
+    if not isinstance(term_freq, dict) or not isinstance(idf, dict):
+        return None
+    if term_freq == {}:
+        return None
+
+    for k, val in term_freq.items():
+        if not isinstance(k, str) or not isinstance(val, float):
+            return None
+        if idf == {}:
+            term_freq[k] = val * math.log(47 / 1)
+        else:
+            for key1, val1 in idf.items():
```

Comment:
You don't need this for - use `idf[key1]`

Reply:
fixed